app/forms/date_form.py

```diff
@@ -87,6 +88,44 @@ class MonthYearDateForm(Form):
     return MonthYearDateForm
 
 
+def get_year_form(answer, answer_store, metadata, error_messages, label, guidance):
+    """
+    Returns a month year form metaclass with appropriate validators. Used in both date and
+    date range form creation.
+
+    :param answer: The answer on which to base this form
+    :param error_messages: The messages to use upon this form during validation
+    :return: The generated MonthYearDateForm metaclass
+    """
+
+    class YearDateForm(Form):
+        year = None
+
+    validate_with = [OptionalForm()]
+
+    if answer['mandatory'] is True:
+        validate_with = validate_mandatory_date(error_messages, answer)
+
+    if 'validation' in answer and 'messages' in answer['validation'] \
+            and 'INVALID_DATE' in answer['validation']['messages']:
+        error_message = answer['validation']['messages']['INVALID_DATE']
+        validate_with.append(YearCheck(error_message))
+    else:
+        validate_with.append(YearCheck())
+
+    if 'minimum' in answer or 'maximum' in answer:
```

Comment:
same as above comment, maybe `has_period_limits`

Reply:
I've put the bespoke message get into it's own method but i'll leave this as it is. Calling it's own method wouldn't actually reduce the lines of code as I would still need to evaluate an if on the return of that method.